### PR TITLE
Update machine template if bundle changes 

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/upgradecluster.go [hunk 1]
```diff
@@ -99,6 +99,7 @@ func (uc *upgradeClusterOptions) upgradeCluster(ctx context.Context) error {
 		deps.ClusterManager,
 		deps.FluxAddonClient,
 		deps.Writer,
+		types.ChangeDiff{},
```
Comment (g-gaston) [inline cmd/eksctl-anywhere/cmd/upgradecluster.go:102]: I don't think we need this as a constructor argument, you can just initialize it in the constructor body
also, I would prefer `types.NewChangeDiff()`

File: pkg/providers/vsphere/vsphere.go [hunk 2]
```diff
@@ -907,14 +907,17 @@ func (p *vsphereProvider) SetupAndValidateDeleteCluster(ctx context.Context) err
...
 	// We plan to approach this way since it's more flexible to add/remove fields and test out for validation
-	if oldC.Spec.KubernetesVersion != newC.Spec.KubernetesVersion {
+	if oldSpec.Cluster.Spec.KubernetesVersion != newSpec.Cluster.Spec.KubernetesVersion {
+		return true
+	}
+	if oldSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host != newSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host {
 		return true
 	}
-	if oldC.Spec.ControlPlaneConfiguration.Endpoint.Host != newC.Spec.ControlPlaneConfiguration.Endpoint.Host {
+	if oldSpec.Bundles.Spec.Number != newSpec.Bundles.Spec.Number {
```
Comment (g-gaston) [inline pkg/providers/vsphere/vsphere.go:920]: Note for other reviewers: this line is the actual important addition. The rest is just renaming

File: pkg/clustermanager/cluster_manager_test.go [hunk 3]
```diff
@@ -896,8 +874,8 @@ type clusterChangedTest struct {
 	oldWorkerMachineConfig, newWorkerMachineConfig             *v1alpha1.VSphereMachineConfig
 }
 
-func newClusterChangedTest(t *testing.T) *clusterChangedTest {
-	testSetup := newTest(t)
+func newClusterChangedTest(t *testing.T, opts ...clustermanager.ClusterManagerOpt) *clusterChangedTest {
```
Comment (g-gaston) [inline pkg/clustermanager/cluster_manager_test.go:877]: nit: Maybe we should rename this method? It was supposed to be only for the `...ClusterChanged` method
but if are finding it useful for other tests, we might want to find a better name
We can do that in a separate PR

**Review by g-gaston — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/527#pullrequestreview-794695993" title="LGTM">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/527#" title="Author self-approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston,taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**